python/ray/autoscaler/_private/commands.py

```diff
@@ -666,7 +666,9 @@ def get_or_create_head_node(config: Dict[str, Any],
         cli_logger.print("Prepared bootstrap config")
 
         if restart_only:
-            setup_commands = []
+            # Some Command Runners may re-launch nodes, requiring setup
+            # commands to be rerun.
+            setup_commands = config["head_setup_commands"]
```

Comment:
Guard this with a docker_in_use

Reply:
is this PR ready for review?

Reply:
@richardliaw yes it is!
^^Follow up from above, I prefer to not guard this with a `docker_in_use` because it violates the abstraction of the NodeUpdater & Command Runner

Reply:
Does that mean that restart_only will rerun head_setup_commands again for non docker as well?

Reply:
setup_commands shouldn't be run if in non-docker mode

Reply:
(that's the whole point of restart-only, that setup commands don't run and only restart ray) -- can you guard this with a config.get("docker") or something?

Reply:
Done!